torchtext/datasets/amazonreviewpolarity.py

```diff
@@ -1,13 +1,15 @@
+from torchtext._internal.module_utils import is_module_available
+
+if is_module_available("torchdata"):
+    from torchdata.datapipes.iter import FileOpener, GDriveReader, IterableWrapper
+
```

Comment:
I wonder if we also need to import `HttpReader` so that we can use this dataset internally within FB by swapping out public URLs with internal URLs. This is what I'm doing for the SST2 dataset implementation (https://github.com/Nayef211/text/blob/main/torchtext/experimental/datasets/sst2.py#L15-L17)

Reply:
That's a good catch @Nayef211. Note that in this case we need GDriveReader. Since this is not yet implementation internally, let fix this part once we import this PR into fb-code.